src/main/java/com/jcraft/jsch/KnownHosts.java

```diff
@@ -425,52 +440,65 @@ void sync() throws IOException {
   }
   synchronized void sync(String foo) throws IOException {
     if(foo==null) return;
-    FileOutputStream fos=new FileOutputStream(Util.checkTilde(foo));
-    dump(fos);
-    fos.close();
+    FileOutputStream fos=null;
+    try {
+      fos=new FileOutputStream(Util.checkTilde(foo));
+      dump(fos);
+    }
+    finally {
+      if (fos != null) {
+        fos.close();
+      }
+    }
   }
 
   private static final byte[] space={(byte)0x20};
-  private static final byte[] cr=Util.str2byte("\n");
+  private static final byte[] lf=Util.str2byte("\n");
+  
+  @SuppressWarnings("unused") // to keep the method's signature without warning
```

Comment:
Curious as to why this is needed since the `out` argument appears to be used in the call to `dumpHostKey()`?

Reply:
I'm not sure I understand why this warning suppression is required?

Reply:
`out` is used but no `IOException` is thrown. You can't remove it from the method signature without potentially breaking things, so my solution was to suppress the warning and keep the method signature as it was before.

Reply:
Why not just remove the `throws IOException` then from the method signature?
It’s a package private method, so it’s not part of the public facing API.

Reply:
I've checked and you're right. Except the testcase I've created this didn't break anything else, so I've changed it